testy/runtime.py

```diff
@@ -1,25 +1,103 @@
-import threading
+import multiprocessing
 import time
-import studentCode
+import sys
+import traceback
+
+import stateManager
+import studentAPI
+
+from runtimeUtil import *
+
+
+# TODO:
+# 0. Set up testing code for the following features.
+# DONE 1. Have student code go through api to modify state.
+# DONE 2. Imposing timeouts on student code (infinite loop, try-catch)
+# 3. Figure out how to kill student thread.
+# 4. Integrate with Bob's socket code: spin up a communication process
+# 5. stateManager throw badThing on processNameNotFound
+# 6. refactor process startup code: higher order function
+# 7. Writeup how all this works
+# 8. Investigate making BadThing extend exception
+
+allProcesses = {}
+badThings = multiprocessing.Condition()
+globalBadThing = "unititialized globalBadThing"
```

Comment:
Rename since it isn't clear what this variable is when initially reading this as you manipulate this variable farther down. Initializing it as a string doesn't help with clarity either.